numpy/core/tests/test_ufunc.py

```diff
@@ -597,6 +597,15 @@ def test_output_argument(self):
         umt.inner1d(a, b, out=c[..., 0])
         assert_array_equal(c[..., 0], np.sum(a*b, axis=-1), err_msg=msg)
 
+        # Passing a temporary array as out= argument emits a Warning.
+        arr = np.ones(3)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
+            np.add(arr, arr, out=np.empty(3))
+        assert len(w) == 1
```

Comment:
Probably needs to be skipped on non-refcounting Pythons (PyPy).

Reply:
You can also just use `assert_warns` (or also `suppress_warnings` since you also match the text). I just like it a bit more:
```
with suppress_warnings() as sup:
    r = sup.record(RutimeWarning, ".*the result may be lost")
    ...
assert_(len(r) == 1)
```
but I do not care much about it.

Reply:
And yes, you have to check for refcounting of the interpreter (I am not sure if the C-Code needs something here).

Reply:
Do we allow `assert` statements nowadays? I something think we might, but we used to not use them (since they are more commonly used with pytest?)